motivaa-backend/motivaa-app/src/test/java/com/motivaa/control/HabitCreationServiceTest.java

```diff
@@ -0,0 +1,263 @@
+package com.motivaa.control;
+
+import com.motivaa.TestUtils.HabitRequestCreator;
+import com.motivaa.control.error_handling.exceptions.FieldCustomValidationException;
+import com.motivaa.control.error_handling.exceptions.NotFoundException;
+import com.motivaa.control.error_handling.exceptions.RepositoryException;
+import com.motivaa.control.repository.MotivaaRepository;
+import com.motivaa.control.utility.MessageBundle;
+import static com.motivaa.control.utility.MessageBundle.INVALID_RECURRING_TYPE_ERROR_MESSAGE;
+import com.motivaa.entity.DaySpecificHabit;
+import com.motivaa.entity.Habit;
+import com.motivaa.entity.NotDaySpecificHabit;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import static org.mockito.ArgumentMatchers.any;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.doNothing;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.Arrays;
+
+@SpringBootTest
+@ActiveProfiles("test")
+public class HabitCreationServiceTest {
+
+    @MockBean
+    private MotivaaRepository motivaaRepository;
+    @MockBean
+    private CommonHabitServices commonHabitServices;
+
+    @Autowired
+    HabitCreationService habitCreationService;
+
+    @Nested
+    class Success_Scenarios {
+
+        @Test
+        void valid_daySpecificHabit_should_be_successful() throws Exception {
```

Comment:
I don't know if you know this but you can build a validator in the test and just pass the object to it, and it will validate it and you can get the validation errors from it:

E.g.: 
 validator = Validation.buildDefaultValidatorFactory().getValidator();
 UserNotNull user = new UserNotNull("John");
 Set<ConstraintViolation<UserNotNull>> violations = validator.validate(user);
 assertThat(violations.size()).isEqualTo(0);
 
 Here is the whole documentation: https://tomee.apache.org/jakartaee-10.0/javadoc/jakarta/validation/executable/ExecutableValidator.html
 
 You can get the validation error message and everything.
 
 If you just want to validate if its valid and not the value specifically you could use this instead of checking for each value.